source/client/process_impl.cc

```diff
@@ -126,6 +127,14 @@ class StatsConfigImpl : public Envoy::Server::Configuration::StatsConfig {
   const std::chrono::milliseconds flush_interval_;
 };
 
+// A fake ServerLifecycleNotifier. Because it does nothing, it's safe to just create
+// multiple instances of it, rather than manage the lifetime of a single one.
+class LifecycleNotifierImpl : public Envoy::Server::ServerLifecycleNotifier {
+public:
+  HandlePtr registerCallback(Stage, StageCallback) override { return nullptr; }
```

Comment:
Are these methods called in Nighthawk's lifecycle? If they are, returning a nullptr could create an unexpected issue for the future and maybe we need another solution.

If they aren't expected to be called, I would recommend we use the same approach as in many other methods that are not supposed to be called - PANIC when they get called. That way we get a much simpler error to debug, i.e. we know that a method started being called and we know where to start investigating.

Reply:
These methods should not be called so putting a PANIC in the method body sounds good.

Reply:
Changed the bodies to PANIC, and the integration tests didn't have any panics.

Reply:
Thank you for looking into it.